src/routes/safe/components/Apps/index.tsx

```diff
@@ -63,11 +60,11 @@ const operations = {
 }
 
 function Apps({ closeModal, closeSnackbar, enqueueSnackbar, openModal }) {
-  const [appList, setAppList] = useState<Array<SafeApp>>([])
-  const [legalDisclaimerAccepted, setLegalDisclaimerAccepted] = useState(false)
-  const [selectedApp, setSelectedApp] = useState<string>()
-  const [loading, setLoading] = useState(true)
-  const [appIsLoading, setAppIsLoading] = useState(true)
+  const { appList, loadingAppList, onAppToggle, onAppAdded } = useAppList()
+
+  const [initialAppSelected, setInitialAppSelected] = useState<boolean>(false)
```

Comment:
I don't understand why are we using this new state. Can't we just use `selectedAppId`?

Reply:
When we load an app, we need to select an app initially, this would be the first app in the list which is not disabled. This variable is used to track if we've already set an app initially so this effect doesn't run in future

Reply:
```
  useEffect(() => {
    const selectFirstEnabledApp = () => {
      const firstEnabledApp = appList.find((a) => !a.disabled)
      if (firstEnabledApp) {
        setSelectedAppId(firstEnabledApp.id)
        setInitialAppSelected(true)
      }
    }

    if (appList.length) {
      // select the app for the first time once the list is loaded
      if (!initialAppSelected) {
        selectFirstEnabledApp()
      }

      // check if the current active app was disabled by the user
      const currentApp = appList.find((app) => selectedAppId === app.id)
      if (currentApp?.disabled) {
        selectFirstEnabledApp()
      }
    }
  }, [appList, initialAppSelected, selectedAppId])
```

this is the hook for selecting first enabled app, as you can see it runs in two cases: initial load or if we disabled previously selected app

Reply:
I think the following could work too.

1st run, AppList, and selectedAppId will be undefined. 
2nd run appList are loaded and selectedAppId === undefined. Here you could set the selectedId (Same indicator as initialAppSelected).

Reply:
hmm makes sense, I think the main point is that selectedAppId should always be defined, I will try to use it for the new implementation. I think I've tried it, but it didn't work. Will give it one more try